pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -249,9 +250,9 @@ func (c *Cluster) Start() error {
 			logger.Debugf("osds prepared on node %s: %+v", n.Name, osds)
 			for _, osd := range osds {
 				logger.Debugf("start osd %v", osd)
-				dp := c.makeOSDDeployment(n.Name, devicesToUse, n.Selection, n.Resources, storeConfig, metadataDevice, n.Location, osd)
-				if dp == nil {
-					errMsg := fmt.Sprintf("nil deployment for node %s due to empty volumes", n.Name)
+				dp, err := c.makeDeployment(n.Name, devicesToUse, n.Selection, n.Resources, storeConfig, metadataDevice, n.Location, osd)
+				if err != nil {
+					errMsg := fmt.Sprintf("nil deployment for node %s: %v", n.Name, err)
```

Comment:
Could you move this `fmt.Sprintf` in the `logger.Warningf` which does the same.

Reply:
the `errMsg` is also used for the orchestration status. Having it in a var in both log and append saves a Sprintf